examples/browser-webrtc/Cargo.toml

```
@@ -0,0 +1,38 @@
+[package]
+authors = ["Doug Anderson <douganderson444@peerpiper.io>"]
+edition = "2021"
+license = "MIT"
+name = "browser-webrtc-example"
+publish = false
+rust-version = { workspace = true }
+version = "0.1.0"
+
+[lib]
+crate-type = ["cdylib"]
+
```

Does it make sense to name the binary differently here in order to satisfy https://github.com/rust-lang/cargo/issues/6313?

```suggestion

[[bin]]
name = "browser-webrtc-example-server" # Name binary differently from library; see https://github.com/rust-lang/cargo/issues/6313
path = "src/main.rs"

```

---

Is this warning showing up? I couldn't see it locally. The crate should only produce one binary because the WASM blob is produced from the library part.

---

It did show up for me yes. Perhaps it has more to do with the names than the number of binaries?

---

Where is the 2nd binary defined though? What am I missing?

---

I can't reproduce that. Just build the entire workspace with `cargo build --all-features --all-targets` without a warning. Which command are you running?

---

I think they mean the **library** name and the **binary** name are the same

It's only a warning for now, but "might break in future releases"

---

I was running `cargo run`

---

Just ran `cargo run` again and got this output:

```bash
The bin target `browser-webrtc-example` in package `browser-webrtc-example v0.1.0 (C:\Users\douga\Documents2\code\RUST-projects\rust-libp2p\examples\browser-webrtc)` has the same output filename as the lib target `browser-webrtc-example` in package `browser-webrtc-example v0.1.0 (C:\Users\douga\Documents2\code\RUST-projects\rust-libp2p\examples\browser-webrtc)`.
Colliding filename is: C:\Users\douga\Documents2\code\RUST-projects\rust-libp2p\target\debug\deps\browser_webrtc_example.pdb
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The bin target `browser-webrtc-example` in package `browser-webrtc-example v0.1.0 (C:\Users\douga\Documents2\code\RUST-projects\rust-libp2p\examples\browser-webrtc)` has the same output filename as the lib target `browser-webrtc-example` in package `browser-webrtc-example v0.1.0 (C:\Users\douga\Documents2\code\RUST-projects\rust-libp2p\examples\browser-webrtc)`.
Colliding filename is: C:\Users\douga\Documents2\code\RUST-projects\rust-libp2p\target\debug\browser_webrtc_example.pdb
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
```

---

I just did a rustup update and can't reproduce this:

```
❯ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.26s
     Running `/home/thomas/src/github.com/libp2p/rust-libp2p/target/debug/browser-webrtc-example`
[2023-09-05T23:22:22Z INFO  browser_webrtc_example] Listening on: /ip6/::1/udp/42713/webrtc-direct/certhash/uEiDsCW4VEcXRRKOKBzsGKcQhSbFL9YW9tuv_Ctb5RBONsw
[2023-09-05T23:22:22Z INFO  browser_webrtc_example] Serving client files at http://127.0.0.1:8080
^C%
```

Is this a Windows-only problem perhaps? This is a completely vanilla Rust crate that follows all conventions. I am very hesitant to change that. Are you sure this isn't something in your setup? I don't know Windows very well but `.pdb` files are for debugging, right? What toolchain are you compiling with? MSVC or GNU?

---

From the issue:

> PDB collisions on Windows. If the package has a binary and a dylib library of the same name, then the .pdb file for each target will have the same name. This is easy to hit if you have a proc-macro package with a binary.

I've just commented there. In my opinion, this is a bug that cargo needs to resolve before they make this a hard error.

---

Ah yes, looks like another Windows specific issue then. I just didn't want it to turn from a warning to an error down the road and give someone a difficult time running the example.

---

I've never seen or used a `.pdb` file so I don't find it useful at all. 

Also, using `nightly-x86_64-pc-windows-msvc`

---

> Ah yes, looks like another Windows specific issue then. I just didn't want it to turn from a warning to an error down the road and give someone a difficult time running the example.

Thank you for that! Much appreciated :)

I do think we can leave this as is for now though.